### PR TITLE
enable Rubocop Style/DateTime rule and remove violations

### PR DESCRIPTION
**Pull request by psinghal20**
Enables Rubocop Style/DateTime rule and change all uses of DateTime class to
Time or Date class.

### CODE REVIEW

**Comment by psinghal20** [comment]
The RSpec tests seem to fail. Can you take a look @ragesoss?

File: .rubocop.yml [hunk 1]
```diff
@@ -48,6 +48,9 @@ Metrics/ModuleLength:
   Exclude:
     - 'spec/**/*' # not a big deal in spec helper module
 
+Style/DateTime:
```
Comment (ragesoss) [inline .rubocop.yml]: Just deleting the part that disables it is enough. Adding `Enabled: true` isn't necessary, as it's on by default.

File: app/models/campaign.rb [hunk 2]
```diff
@@ -100,7 +100,7 @@ def validate_dates
   # Intercept Rails typecasting and add error if given value cannot be parsed into a date.
   def validate_date_attribute(date_type)
     value = send("#{date_type}_before_type_cast")
-    self[date_type] = value.is_a?(Date) || value.is_a?(Time) ? value : DateTime.parse(value)
+    self[date_type] = value.is_a?(Date) || value.is_a?(Time) ? value : Date.parse(value)
```
Comment (ragesoss) [inline app/models/campaign.rb]: We should probably use Time here, as it's more directly a substitute for DateTime.

**Comment by ragesoss** [comment]
@psinghal20 this looks good to go once the comments are addressed.

File: app/models/campaign.rb (hunk 2 above)
Comment (psinghal20) [reply app/models/campaign.rb]: Hi, the parse function we are using here is not available in Time class in ruby-2.5.0, as far as I know, I tried searching for it but couldn't find an equivalent function and thus went with Date instead.

Comment (ragesoss) [reply app/models/campaign.rb]: It's there: https://ruby-doc.org/stdlib-2.5.0/libdoc/time/rdoc/Time.html#method-c-parse

If you're working in `irb`, then you won't get that functionality unless you `require 'time'`. But in Rails, it looks like it's already loaded and will just work.

**Pull request merged**